mpqp/core/circuit.py

```diff
@@ -1280,12 +1297,15 @@ def __str__(self) -> str:
 
     def __repr__(self) -> str:
         instructions_repr = ", ".join(repr(instr) for instr in self.instructions)
+        nb_qubits = ""
+        if self._user_nb_qubits is not None:
```

Comment:
If you call the property `self.nb_qubits`, then this check is already done there, so you can directly use the property for the repr no ?